service/src/test/java/uk/gov/hmcts/reform/fpl/service/DocumentUploadHelperTest.java

```diff
@@ -0,0 +1,92 @@
+import uk.gov.hmcts.reform.fpl.request.RequestData;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+import java.util.Arrays;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {
+    DocumentUploadHelper.class
+})
+class DocumentUploadHelperTest {
+    private static final String AUTH_TOKEN = "token";
+    private static final String USER_ID = "1";
+
+    @MockBean
+    private IdamClient idamClient;
+
+    @MockBean
+    private RequestData requestData;
+
+    @Autowired
+    private DocumentUploadHelper documentUploadHelper;
+
+    @BeforeEach
+    void setUp() {
+        when(requestData.authorisation()).thenReturn(AUTH_TOKEN);
+    }
+
+    @Test
+    void shouldReturnUploadedDocumentUserInfoForUserWithHmctsRole() {
+        when(idamClient.getUserDetails(eq(AUTH_TOKEN))).thenReturn(createUserDetailsWithHmctsRole());
+        assertThat(documentUploadHelper.getUploadedDocumentUserDetails()).isEqualTo("HMCTS");
+    }
+
+    @Test
+    void shouldReturnUploadedDocumentUserInfoForUserWithNonHmctsRole() {
+        when(idamClient.getUserDetails(eq(AUTH_TOKEN))).thenReturn(createUserDetailsWithNonHmctsRole());
+        assertThat(documentUploadHelper.getUploadedDocumentUserDetails()).isEqualTo("steve.hudson@gov.uk");
+    }
+
+    @Test
+    void shouldReturnUploadedDocumentUserInfoForUserWithHmctsAndNonHmctsRole() {
+        when(idamClient.getUserDetails(eq(AUTH_TOKEN))).thenReturn(createUserDetailsWithHmctsAndNonHmctsRole());
+        assertThat(documentUploadHelper.getUploadedDocumentUserDetails()).isEqualTo("HMCTS");
+    }
+
+    private UserDetails createUserDetailsWithHmctsRole() {
+        return UserDetails.builder()
+            .id(USER_ID)
+            .surname("Hudson")
+            .forename("Steve")
+            .email("steve.hudson@gov.uk")
+            .roles(Arrays.asList("caseworker-publiclaw-courtadmin", "caseworker-publiclaw-judiciary"))
+            .build();
+    }
+
+    private UserDetails createUserDetailsWithNonHmctsRole() {
+        return UserDetails.builder()
+            .id(USER_ID)
+            .surname("Hudson")
+            .forename("Steve")
+            .email("steve.hudson@gov.uk")
+            .roles(Arrays.asList("caseworker-publiclaw-solicitor", "caseworker-publiclaw-cafcass"))
+            .build();
+    }
+
+    private UserDetails createUserDetailsWithHmctsAndNonHmctsRole() {
+        return UserDetails.builder()
+            .id(USER_ID)
+            .surname("Hudson")
+            .forename("Steve")
+            .email("steve.hudson@gov.uk")
+            .roles(Arrays.asList(
+                "caseworker-publiclaw-solicitor",
+                "caseworker-publiclaw-cafcass",
+                "caseworker-publiclaw-superuser"))
+            .build();
+    }
```

Comment:
could be refactored to `UserDetails createUserDetails(List<String> roles)`

Reply:
I want name to specifically say user has both HMCTS and NonHMCTS role. I am guessing that is a possibility.

Reply:
you could do that in the variable itself no?